File: src/main/java/com/pancisin/bookster/models/Programme.java
package com.pancisin.bookster.models;

import java.sql.Time;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonIgnore;

@Entity
@Table(name = "programmes")
public class Programme {

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;
	
	@Column
	private Time time;

	@Column
	private Time duration;
	
	@Column
	private String name;
	
	@Column
	private String description;

	@JsonIgnore
	@ManyToOne
	private Event event;
	
	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Time getTime() {
		return time;
	}

	public void setTime(Time time) {
		this.time = time;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public Event getEvent() {
		return event;
	}

	public void setEvent(Event event) {
		this.event = event;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public Time getDuration() {
		return duration;
	}

	public void setDuration(Time duration) {
		this.duration = duration;
	}
}

File: src/main/kotlin/com/pancisin/bookster/model/Programme.kt
package com.pancisin.bookster.model

import java.sql.Time

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.ManyToOne
import javax.persistence.Table
import javax.validation.constraints.NotNull

import com.fasterxml.jackson.annotation.JsonIgnore

@Entity
@Table(name = "programmes")
data class Programme(
  @Id @GeneratedValue(strategy = GenerationType.AUTO)
  var id: Long? = null,

  @NotNull @Column
  var time: Time? = null,

  @Column
  var duration: Time? = null,

  @NotNull @Column
  var name: String? = null,

  @Column
  var description: String? = null,

  @JsonIgnore @ManyToOne
  var event: Event? = null
)
